hub/core/dataset/__init__.py

```diff
@@ -21,5 +20,8 @@ def dataset_factory(path, *args, **kwargs):
         clz = Dataset
 
     if clz in {Dataset, HubCloudDataset}:
-        return clz(path=path, *args, **kwargs)
+        ds = clz(path=path, *args, **kwargs)
+        if "VDS_INDEX" in ds.tensors:
```

Comment:
This isn't a best way to detect if it is view. Better to check meta-data.

Reply:
There is a weird bug when checking info here.